src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -106,6 +105,25 @@
 
 	private static final String FIND_GERMPLASM_BY_GIDS = " g.gid in (:gids) ";
 
+	private static final String FIND_GERMPLASM_WITHDESCENDANTS = "SELECT g.gid as gid FROM germplsm g "
+			+ "WHERE (EXISTS (SELECT 1 FROM germplsm descendant WHERE g.gid = descendant.gpid1 OR g.gid = descendant.gpid2 ) "
+			+ "OR EXISTS (SELECT 1 FROM progntrs p WHERE  g.gid = p.gid)) "
+			+ "AND g.gid IN (:gids) AND  g.deleted = 0 AND g.grplce = 0 ";
+
+	private static final String FIND_GERMPLASM_WITH_DERIVATIVE_OR_MAINTENANCE_DESCENDANTS = "SELECT g.gid as gid FROM germplsm g "
+			+ "WHERE (EXISTS "
+						+ " (SELECT 1 FROM germplsm descendant "
+						+ " INNER JOIN methods mtype ON mtype.mid = descendant.methn"
+						+ " WHERE (g.gid = descendant.gpid1 OR g.gid = descendant.gpid2) "
+						+ " AND mtype.mtype IN (:mtypes) ) "
+						+ " OR EXISTS "
+						+ " (SELECT 1 FROM progntrs p "
```

Comment:
I think we don't have to search if the germplasm is used in the progntrs table. We only have to check if the germplasm is used in gpid1 or gpid2, since we are just looking for a der/man germplasm.